janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -745,29 +807,54 @@ private String getDeletionScript(KeyInformation.IndexRetriever information, Stri
                 case LIST:
                     final String jsValue = convertToJsType(deletion.value, compat.scriptLang(), Mapping.getMapping(keyInformation));
                     String index = INDEX_NAME + i++;
-                    script.append("def ")
+                    script.append("if (ctx._source.containsKey(\"")
```

Comment:
Could you briefly explain the changes to this method? I'm not sure why they are necessary to support parametrized scripts as this method is only used for scripts that cannot be parametrized.

Reply:
Here I just added few guards for NPEs that we faced in production. And yes you're right this code won't be used in parametrized scripts but I added them just in case if we decided to switch parameterization off someday.

Reply:
Did you add any tests here that produce the NPEs without your changes? If not, I can't really verify the changes.
I would suggest that you exclude these changes and create a separate issue for them where you describe the problem you saw in production. That will make it easier to address the problem in a separate PR.

Reply:
Yes, I've added tests, they are:
- `testListDelete`
- `testRemoveNonExistentProperty`
I'll file them as a separate PR.